### PR TITLE
chore: switch from tslint to eslint

### LINKED ISSUE #305: ESLint + Prettier
Would you mind if I added some basic linting and possibly Prettier to the UI setup, as git/npm hooks? It would help keep things consistent and help prevent issues from future contributors (myself included).

### PR DESCRIPTION
**Pull request by zacanger**
* Removes TSLint config
* Adds ESLint config, with several rules disabled (some of these disables should be removed eventually, I just didn't want to touch the actual code here)
* Fixes one parsing error (`xmlns:xlink`) so that ESLint passes

This does not bring in Sortpack, Prettier, Husky, or Lint-Staged.

Closes #305

### CODE REVIEW

File: ui/src/components/symbols.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ export class Symbols extends Component<any, any> {
 
   render() {
     return (
-      <svg aria-hidden="true" style="position: absolute; width: 0; height: 0; overflow: hidden;" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+      <svg aria-hidden="true" style="position: absolute; width: 0; height: 0; overflow: hidden;" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
```
Comment (zacanger) [inline ui/src/components/symbols.tsx]: That one attribute is invalid JSX. I think it just got stripped out when compiled, but ESLint chokes on it.

Comment (dessalines) [reply ui/src/components/symbols.tsx]: Oh sweet, thx.

**Comment by dessalines** [comment]
I'm not opposed to bringing those 4 things in too, unless there's some reason I should add them instead(since husky might force you to fix everything before you commit)

**Comment by zacanger** [comment]
Sweet, I added Husky, Lint-Staged, and Sortpack back in. The problem with bringing in Prettier (at least, the way I had it in the other branch, integrating it with ESLint) is that it'll format everything on precommit and block push on formatting that doesn't meet the config.

**Comment by dessalines** [comment]
I gotcha, I can add that, from looking at your other one. I'll merge this shortly. Thanks! I had no idea that eslint was so good now.

**Pull request merged**